lib/utils.ts

```diff
@@ -111,3 +112,18 @@ export function promiseWithTimeout<T>(ms: number, promise: Promise<T>): Promise<
     });
   });
 }
+
+
+/** Finds an exe file in the package assuming it is placed under `rootPath/platform-arch/exe`
+ * For example on Windows x64, if the `exeName` is `serve-d`, it returns the absolute path to `./bin/win32-x64/exeName.exe`
+ * @param exeName name of the exe file
+ * @param rootPath the path of the folder of the exe file. Defaults to 'bin'
+ * @param exeExtention the extention of the exe file. Defaults to `process.platform === "win32" ? ".exe" : ""`
+ */
+export function getExePath(
+ exeName: string,
+ rootPath = "bin",
+ exeExtention = process.platform === "win32" ? ".exe" : ""
+): string {
+ return resolve(join(rootPath, `${process.platform}-${process.arch}`, `${exeName}${exeExtention}`));
```

Comment:
should `` `${process.platform}-${process.arch}` `` be configurable? maybe as part of `rootPath` like

```diff
- rootPath = "bin",
+ rootPath = join("bin", `${process.platform}-${process.arch}`),
```

Reply:
I almost feel like this doesn't need to be a utility function since it isn't much easier than writing 

```js
resolve(join(rootPath, `${process.platform}-${process.arch}`, `serve-d${process.platform === "win32" ? ".exe" : ""}`))
```

And that is infinitely more configurable than this utility function.

Reply:
Yes. Fixed in https://github.com/atom-community/atom-languageclient/pull/124/commits/9746992e9a358815e861cbc48753e52d74fe45c1